src/core/GodotGlobal.cpp

```diff
@@ -14,12 +15,54 @@ void Godot::print(const String& message)
 
 void Godot::print_warning(const String& description, const String& function, const String& file, int line)
 {
-	godot::api->godot_print_warning(description.c_string(), function.c_string(), file.c_string(), line);
+	int len;
+
+	description.get_c_string(NULL, &len);
+	char * c_desc = (char *)godot::api->godot_alloc(len + 1);
+	description.get_c_string(c_desc, &len);
+	c_desc[len] = '\0';
+
+	function.get_c_string(NULL, &len);
+	char * c_func = (char *)godot::api->godot_alloc(len + 1);
+	function.get_c_string(c_func, &len);
+	c_func[len] = '\0';
+
+	file.get_c_string(NULL, &len);
+	char * c_file = (char *)godot::api->godot_alloc(len + 1);
+	file.get_c_string(c_file, &len);
+	c_file[len] = '\0';
```

Comment:
Haven't used the cpp bindings yet, but seems like those lines might be optimized into some `alloc_c_string` function, no?

Reply:
yeah could do that, I went for the quick and dirty here. Bit annoying to even have to go down this route.